src/Components/Patient/PatientRegister.tsx

```diff
@@ -218,6 +218,16 @@ export const PatientRegister = (props: PatientRegisterProps) => {
     }
   }, [careExtId, formField]);
 
+  useEffect(() => {
+    const showAllFacilityUsers = ["DistrictAdmin", "StateAdmin"];
+    if (
+      !showAllFacilityUsers.includes(authUser.user_type) &&
+      authUser.home_facility_object?.id !== facilityId
+    ) {
+      navigate("/facility");
+    }
+  }, []);
```

Comment:
Redirecting on mount doesn't seem to be a nice idea. It'd lead to weird behaviour if user is not allowed to enter the page. If at all user accidentally enters a page that user is not permitted to visit, we should show page not found instead.


Also, are you sure the authzn checks are correct?